app/test/ui/screens/create_post/create_post_controller_impl_test.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:medi_support/ui/screens/create_post/create_post_controller_impl.dart';
+import 'package:medi_support/ui/screens/create_post/services/create_post_backend_service.dart';
+import 'package:medi_support/ui/screens/create_post/services/create_post_navigation_service.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'create_post_controller_impl_test.mocks.dart';
+
+@GenerateMocks(<Type>[
+  CreatePostNavigationService,
+  CreatePostBackendService,
+])
+class CreatePostControllerImplTest {}
+
+void main() {
+  late MockCreatePostBackendService mockCreatePostBackendService;
+  late MockCreatePostNavigationService mockCreatePostNavigationService;
+
+  setUp(
+    () {
+      mockCreatePostBackendService = MockCreatePostBackendService();
+      mockCreatePostNavigationService = MockCreatePostNavigationService();
+    },
+  );
+
+  CreatePostControllerImpl createCreatePostController() {
+    final CreatePostControllerImpl createPostController =
+        ProviderContainer().read(
+      createPostControllerImplProvider(
+        backendService: mockCreatePostBackendService,
+        navigationService: mockCreatePostNavigationService,
+      ).notifier,
+    );
+    final KeepAliveLink keepAliveLink = createPostController.ref.keepAlive();
+    addTearDown(keepAliveLink.close);
+    return createPostController;
+  }
+
+  test(
+    'CreatePostControllerImpl should be able to send a post',
+    () {
+      createCreatePostController().send(
+        title: 'Title',
+        body: 'Body',
+        onSend: () {},
+      );
+      verify(
+        mockCreatePostBackendService.createPost(
+          title: 'Title',
+          content: 'Body',
+        ),
+      ).called(1);
+    },
+  );
+
+  test(
```

Comment:
**nitpick:** I think it would be nicer to split into two tests, but that is up to you.

Reply:
good idea